sdk/openai/inference/examples/azure_chat_completions.rs

```diff
@@ -0,0 +1,45 @@
+use azure_core::Result;
+use azure_openai_inference::{
+    clients::{AzureOpenAIClient, AzureOpenAIClientMethods, ChatCompletionsClientMethods},
+    request::CreateChatCompletionsRequest,
+    AzureOpenAIClientOptions, AzureServiceVersion,
+};
+
+// This example illustrates how to use Azure OpenAI with key credential authentication to generate a chat completion.
+#[tokio::main]
+pub async fn main() -> Result<()> {
```

Comment:
Nit: examples tend to use `unwrap()` or `expect()` so the error cases are obvious.